krystex/src/main/java/com/flipkart/krystal/krystex/kryon/AbstractKryon.java

```diff
@@ -69,10 +70,15 @@ protected NavigableSet<MainLogicDecorator> getSortedDecorators(DependantChain de
                 kryonId,
                 mainLogicDefinition.logicTags(),
                 dependantChain,
-                kryonDefinition.kryonDefinitionRegistry())));
+                definition.kryonDefinitionRegistry())));
     TreeSet<MainLogicDecorator> sortedDecorators =
         new TreeSet<>(logicDecorationOrdering.decorationOrder());
     sortedDecorators.addAll(decorators.values());
     return sortedDecorators;
   }
+
+  @Override
+  public KryonDefinition getDefinition() {
+    return definition;
```

Comment:
Change this field back to `kryonDefinition` and method also to getKryonDefinition to avoid so much diffs